api/v1alpha1/spacerequest_types.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SpaceRequestSpec defines the desired state of Space
+// +k8s:openapi-gen=true
+type SpaceRequestSpec struct {
+	// TierName is a required property introduced to retain the name of the tier
+	// for which this Space is provisioned
+	// If not set then the tier name will be set automatically
+	// +optional
+	TierName string `json:"tierName,omitempty"`
+
+	// TargetClusterLabels one or more labels that define a set of clusters
+	// where the Space can be provisioned.
+	// +optional
+	TargetClusterLabels map[string]string `json:"targetClusterLabels,omitempty"`
+}
+
+// SpaceRequestStatus defines the observed state of Space
+// +k8s:openapi-gen=true
+type SpaceRequestStatus struct {
+
+	// TargetClusterURL The API URL of the cluster where Space is currently provisioned
+	// Can be empty if provisioning did not start or failed
+	// To be used to de-provision the NSTemplateSet if the Spec.TargetCluster is either changed or removed
+	// +optional
+	TargetClusterURL string `json:"targetClusterURL,omitempty"`
+
+	// The status of user accounts in the member clusters which belong to this MasterUserRecord
+	// +listType=atomic
+	NamespaceAccess []NamespaceAccess `json:"namespaceAccess,omitempty"`
```

Comment:
let's mark it as optional
```suggestion
	// +optional
	NamespaceAccess []NamespaceAccess `json:"namespaceAccess,omitempty"`
```

Reply:
done in : [04797c2d7449e7](https://github.com/codeready-toolchain/api/pull/333/commits/04797c2d7449e763fe1aa2f83a3058b5951df2c8)